extras/check_code_style.sh

```diff
@@ -1,279 +1,212 @@
 
-    exit $1
-}
+    --[i]ncremental             Prompt user before each file analysis.
 
-#
-# bail_out message
-#
-bail_out ()
-{
-    echo "$1"
-    exit 1
+    --file=/path/to/file        Perform the analysis on this file.
+
+    --git-start=Git_SHA_value   Hash value (Git SHA) from which Git starts the diff.
+                                Default: --git-start=master
+
+    --git-end=Git_SHA_value     Hash value (Git SHA) at which Git ends the diff.
+                                Default: --git-start=HEAD
+
+    --cppcheck=exe              The name of the CPPCHECK executable.
+                                Default: --cppcheck=cppcheck
+                                Note: CPPCHECK version 1.69 or later is required.
+                                      This corresponds to the version installed in
+                                      the NEST Travis CI build and test environment.
+
+    --clang-format=exe          The name of the CLANG-FORMAT executable.
+                                Default: --clang-format=clang-format-3.6
+                                Note: CLANG-FORMAT version 3.6 is required.
+                                      This corresponds to the version installed in
+                                      the NEST Travis CI build and test environment.
+
+    --vera++=exe                The name of the VERA++ executable.
+                                Default: --vera++=vera++
+
+    --pep8=exe                  The name of the PEP8 executable.
+                                Default: --pep8=pep8
+EOF
 }
 
+echo ""
+echo "+ + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + +"
+echo "+                       NEST STATIC CODE ANALYSIS TOOL                        +"
+echo "+ + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + +"
+echo ""
+
+# Process parameters.
 while test $# -gt 0 ; do
-    case "$1" in
-        --help)
-            usage 0
-            ;;
-        --incremental)
-            INCREMENTAL=true
-            ;;
-        --file=*)
-            ONLY_FILE="$( echo "$1" | sed 's/^--file=//' )"
-            if [ ! -f $ONLY_FILE ]; then
-              bail_out "This is not a valid file: $ONLY_FILE"
-            fi
-            ;;
-        --git-start=*)
-            START_SHA="$( echo "$1" | sed 's/^--git-start=//' )"
-            ;;
-        --git-end=*)
-            END_SHA="$( echo "$1" | sed 's/^--git-end=//' )"
-            ;;
-        --cppcheck=*)
-            CPPCHECK="$( echo "$1" | sed 's/^--cppcheck=//' )"
-            ;;
-        --vera++=*)
-            VERA="$( echo "$1" | sed 's/^--vera++=//' )"
-            ;;
-        --pep8=*)
-            PEP8="$( echo "$1" | sed 's/^--pep8=//' )"
-            ;;
-        --clang-format=*)
-            CLANG_FORMAT="$( echo "$1" | sed 's/^--clang-format=//' )"
-            ;;
-        *)
-            usage 1 "$1"
-            ;;
-    esac
-    shift
+  case "$1" in
+      --help)
+          print_usage
```

Comment:
This should call `usage`, so that the script is exited after help has been printed.